core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -125,19 +126,26 @@ private[spark] class MetricsSystem private (
    *         application, executor/driver and metric source.
    */
   private[spark] def buildRegistryName(source: Source): String = {
-    val appId = conf.getOption("spark.app.id")
+    val metricsNamespace = conf.get(METRICS_NAMESPACE).map(Some(_))
```

Comment:
Since you'd rather not change the constant, this is better written as:

    conf.get(METRICS_NAMESPACE).orElse(conf.getOption("spark.app.id"))